### PR TITLE
Update mongo documents for new channels model

### PR DESCRIPTION
**Pull request by mhilton**
Update the documents and fix some tests that subsequently failed.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1541/
Test PASSed.

File: internal/mongodoc/doc.go [hunk 1]
```diff
@@ -182,10 +185,14 @@ type BaseEntity struct {
 	// The permissions apply to all revisions.
 	ACLs ACL
```
Comment (frankban) [inline internal/mongodoc/doc.go:186]: We could mentions that ACLs apply to unpublished entities, not assigned a stable or development channel.

File: internal/storetesting/entities.go [hunk 2]
```diff
@@ -127,5 +127,23 @@ func AssertBaseEntity(c *gc.C, db *mgo.Collection, expect *mongodoc.BaseEntity)
 	var baseEntity mongodoc.BaseEntity
 	err := db.FindId(expect.URL).One(&baseEntity)
 	c.Assert(err, gc.IsNil)
-	c.Assert(&baseEntity, jc.DeepEquals, expect)
+	c.Assert(&baseEntity, jc.DeepEquals, NormalizeBaseEntity(expect))
+}
+
+// NormalizeBaseEntity modifies a base entity so that it can be compared
+// with a base entity retrieved from mongodb using jc.DeepEquals. If
+// either StableSeries or Development series are nil maps then they will
```
Comment (frankban) [inline internal/storetesting/entities.go]: s/Development series/DevelopmentSeries/

**Comment by frankban** [comment]
:+1: thank you!

**Comment by urosj** [comment]
:+1: We could be more explicit about a map of pointers with DevelopmentSeries and StableSeries in the name, but let's leave it and reconsider later.

File: internal/storetesting/entities.go (hunk 2 above)
Comment (mhilton) [inline internal/storetesting/entities.go]: done

File: internal/mongodoc/doc.go (hunk 1 above)
Comment (mhilton) [inline internal/mongodoc/doc.go:186]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1542/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**